src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java

```diff
@@ -166,33 +177,48 @@ private void mapStatus(final LabResult labResult, final Observation observation)
-
-                // Observation.value.valueQuantity.unit = SG18.RSL.C848.6410
-                quantity.setUnit(investigationResult.getMeasurementUnit());
-
-                // Observation.value.valueQuantity.comparator = SG18.RSL.C830(1).6321
-                investigationResult.getMeasurementValueComparator()
-                    .map(MeasurementValueComparator::getDescription)
-                    .map(QuantityComparator::fromCode)
-                    .ifPresent(quantity::setComparator);
-
-                observation.setValue(quantity);
+                if (investigationResult.getResultType() == LaboratoryInvestigationResultType.NUMERICAL_VALUE) {
+                    final var quantity = new Quantity();
+
+                    // Observation.value.valueQuantity.value = SG18.RSL.C830(1).6314
+                    quantity.setValue(investigationResult.getMeasurementValue());
+
+                    // Observation.value.valueQuantity.unit = SG18.RSL.C848.6410
+                    quantity.setUnit(investigationResult.getMeasurementUnit());
+
+                    // Observation.value.valueQuantity.comparator = SG18.RSL.C830(1).6321
+                    investigationResult.getMeasurementValueComparator()
+                        .map(MeasurementValueComparator::getDescription)
+                        .map(QuantityComparator::fromCode)
+                        .ifPresent(quantity::setComparator);
+
+                    observation.setValue(quantity);
+                } else if (investigationResult.getResultType() == LaboratoryInvestigationResultType.CODED_VALUE) {
+                    final Coding coding = new Coding()
+                        .setCode(investigationResult.getCode())
+                        .setDisplay(investigationResult.getDescription());
+                    investigationResult.getCodingType().ifPresent(
+                        codingType -> coding.setSystem(getSystemValue(codingType))
+                    );
+
+                    CodeableConcept result = new CodeableConcept().addCoding(coding);
+
+                    observation.setValue(result);
+                }
             });
         }
 
         private void mapCode(final LabResult labResult, final Observation observation) {
             // Observation.code = SG18.INV.C847.9930 and SG18.INV.C847.9931
             final var coding = observation.getCode().addCoding();
-            labResult.getInvestigation().getInvestigationCode().ifPresent(coding::setCode);
-            coding.setDisplay(labResult.getInvestigation().getInvestigationDescription());
-            coding.setSystem(CODING_SYSTEM);
+            labResult.getInvestigation().getCode().ifPresent(coding::setCode);
+            coding.setDisplay(labResult.getInvestigation().getDescription());
+            labResult.getInvestigation().getCodingType().ifPresent(
+                codingType -> coding.setSystem(getSystemValue(codingType))
+            );
```

Comment:
The team has had some disagreement about is "simpler", or "easier to read" here, but here's my preference:
```suggestion
            labResult.getInvestigation().getCodingType()
                .map(this::getSystemValue)
                .ifPresent(coding::setSystem);
```

Reply:
I'm happy with both. Think your way is more common across the codebase so I'll implement it like that